test/testdrive/copy-to-s3-minio.td

```diff
@@ -138,3 +138,24 @@ $ s3-verify-data bucket=copytos3 key=test/2
 
 $ s3-verify-data bucket=copytos3 key=test/3
 "{1,2}",f,Infinity,"{""s"":""abc""}",1,32767,2147483647,9223372036854775807,12345678901234567890123.4567890123456789,2010-10-10,10:10:10,2010-10-10 10:10:10,00:00:00,aaaa,\x5c7841414141,това е,\xd182d0b5d0bad181d182
+
+#
+# Copy a large amount of data and force the copy to timeout to see that the INCOMPLETE
+# sentinel file is left behind
+#
+
+$ set-sql-timeout duration=1s force=true
+$ set-max-tries max-tries=1
+
+! COPY (SELECT * FROM generate_series(1, 5000000)) TO 's3://copytos3/test/5'
+  WITH (
+    AWS CONNECTION = aws_conn,
+    MAX FILE SIZE = "100MB",
+    FORMAT = 'csv'
+  );
+timeout
+
+$ set-sql-timeout duration=60s
+$ set-max-tries max-tries=20
+
+$ s3-verify-incomplete bucket=copytos3 key=test/5
```

Comment:
The raciness of this makes me nervous. I have a proposal for how to make it less racy. The `$ postgres-execute` command could learn to take a `background=true` parameter, which kicks off the execution of a SQL statement but doesn't block on its completion. Then you wouldn't need the timeout; you could just let the internal retries in `$ s3-verify ...` wait for the INCOMPLETE file to get created.

Reply:
good point - I've reworked the test to use your suggestion (a new `background` parameter on `postgres-execute`)